### DISCUSSION TITLE
[✨] Skip q-data.json fetches where possible

### DISCUSSION THREAD

**Discussion by cmbartschat** [QwikDev/qwik-evolution#154] [Proposals For Qwik]
### Is your feature request related to a problem?

It seems that currently, any navigation (including changing query params), will trigger a q-data.json fetch. 

There is good reason to do this (even if there are no loaders), for example if there is a plugin/middleware that might redirect.

However, when there are no loaders, plugins, or middleware, it seems like skipping the q-data.json fetch when navigating within an "known path" should not re-fetch.

This can be seen on [this stackblitz](https://stackblitz.com/edit/qwik-starter-1ijkny?file=src%2Froutes%2Findex.tsx) - switching the active file is stored as a query param, but that forces us to continually re-fetch q-data.json from the server which adds latency to rendering.

### Describe the solution you'd like

One potential solution is for q-data.json to return an extra bit indicating "this path is completely inert", and the framework on the client will see that and know that there's no need to ever refetch it, even if a query param changed. It would be nice if it worked for wildcard routes as well.

### Describe alternatives you've considered

Another alternative is to use the browser-level history APIs it doesn't force a q-data fetch. 

[Stackblitz showing the workaround](https://stackblitz.com/edit/qwik-starter-1fmah8?file=src%2Froutes%2Findex.tsx)

The code is slightly uglier but it's still server-side rendered at least, only problem is that it makes the back button into a full reload rather than client-side. Maybe there's a further improvement there. If this can be solved in user space with a convenience hook like `useClientLocation`/`useClientNavigate`, maybe this isnt a huge deal but yeah.

**Comment by bolt-new-by-stackblitz** [comment]
<a href='https://stackblitz.com/~/github.com/BuilderIO/qwik/issues/4226?repros=qwik-starter-1ijkny,qwik-starter-1fmah8'><img src='https://developer.stackblitz.com/img/fix_this_issue_small.svg' alt='Fix this issue in StackBlitz Codeflow' align='left' width='117' height='20'></a> _Start a new pull request in [StackBlitz Codeflow](https://stackblitz.com/~/github.com/BuilderIO/qwik/issues/4226?repros=qwik-starter-1ijkny,qwik-starter-1fmah8)._

- [**qwik-starter-1ijkny** (Open reproduction)](https://stackblitz.com/edit/qwik-starter-1ijkny?issueRepo=BuilderIO/qwik&issueNumber=4226)
- [**qwik-starter-1fmah8** (Open reproduction)](https://stackblitz.com/edit/qwik-starter-1fmah8?issueRepo=BuilderIO/qwik&issueNumber=4226)

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by shairez** [comment]
thanks for the improvement request
we discussed it and agreed it's important to add, but after we'll finish some more urgent tasks first

thanks again!

**Comment by wmertens** [comment]
closing in favor of #246
